### ISSUE TITLE
Error when using @ProvideReactive: Cannot redefine property

### ISSUE THREAD

**Issue by PeterPanZH** [kaorun343/vue-property-decorator#277]
When I upgrade vue-property-decorator to 8.3.0, it comes with a `TypeError: Cannot redefine property` on line `Object.defineProperty(rv[reactiveInjectKey], provide.managedReactive[i], {`.
I think it was related to #249 and caused by #264 .

```
const InjectKey = Symbol();

@Component
class ParentComponent extends Vue {
    @ProvideReactive(InejctKey) foo: any = {};
}

@Component
class ChildComponent extends Vue {
    @InjectReactive({
        from: InejctKey,
        default: ({})
    })
    readonly foo!: any;
}
```

**Comment by fluidsonic** [comment]
I just ran into the same issue. When a second instance of a component that uses `@ProvideReactive` is being created `TypeError: Cannot redefine property` is thrown.

Downgrading to 8.2.2 fixes the issue.

**Comment by gmoneh** [comment]
Yes. The ProvideReactive/InjectReactive feature is broken in 8.3.0 after this change. In my child components, I get:
Injection "__reactiveInject__" not found

**Comment by Micene09** [comment]
any news here? I'm getting this error too.

**Comment by Mrminfive** [comment]
set `inject: []` in Component options

``` ts
const InjectKey = Symbol();

@Component({ inject: [] })
class ParentComponent extends Vue {
    @ProvideReactive(InejctKey) foo: any = {};
}

@Component
class ChildComponent extends Vue {
    @InjectReactive({
        from: InejctKey,
        default: ({})
    })
    readonly foo!: any;
}
```
 
because

``` ts
// inject parent reactive services (if any)
if (!Array.isArray(componentOptions.inject)) {
    componentOptions.inject = componentOptions.inject || {};
    componentOptions.inject[reactiveInjectKey] = { from: reactiveInjectKey, default: {}};
}
```

**Comment by PeterPanZH** [comment]
@Mrminfive That works in 8.3.0. For a temporarily solution.

**Comment by Jonny-china** [comment]
@Mrminfive thanks!

**Comment by LYnnProspery** [comment]
> set `inject: []` in Component options
> 
> ```ts
> const InjectKey = Symbol();
> 
> @Component({ inject: [] })
> class ParentComponent extends Vue {
>     @ProvideReactive(InejctKey) foo: any = {};
> }
> 
> @Component
> class ChildComponent extends Vue {
>     @InjectReactive({
>         from: InejctKey,
>         default: ({})
>     })
>     readonly foo!: any;
> }
> ```
> 
> because
> 
> ```ts
> // inject parent reactive services (if any)
> if (!Array.isArray(componentOptions.inject)) {
>     componentOptions.inject = componentOptions.inject || {};
>     componentOptions.inject[reactiveInjectKey] = { from: reactiveInjectKey, default: {}};
> }
> ```

it will lead the injects in component get undefined

**Comment by tsofist** [comment]
Any updates for this issue?

> it will lead the injects in component get undefined

**This (_new_) problem affected from `v8.4.0`**

I'm rollback to `v8.3.0` 😞

**Comment by redakzter01** [comment]
> I just ran into the same issue. When a second instance of a component that uses `@ProvideReactive` is being created `TypeError: Cannot redefine property` is thrown.
> 
> Downgrading to 8.2.2 fixes the issue.

Same here again ...with version 9.0.0 😕

**Comment by Joebayld** [comment]
I'm having this issue on v8.5.1. Is everyone still just using 8.3.0 for now? 

I can look into a PR if anyone has insight as to what exactly broke..

**Comment by GrzegorzKazana** [comment]
FYI, this would be fixed in PR #330.

**Comment by bodograumann** [comment]
Same problem here on 9.0.0.
This makes me wonder. Could I use the following as a replacement?

```
@Provide() foo = Vue.observable({ key: value });
```

**Comment by rodsscg** [comment]
Same problem here (v.8.4.0)

**Comment by spb-web** [comment]
Same problem here on 8.5.1

**Comment by iScorpius** [comment]
Same problem here (v.8.5.2).
How terrible is it.

**Comment by iScorpius** [comment]
> Same problem here on 9.0.0.
> This makes me wonder. Could I use the following as a replacement?
> 
> ```
> @Provide() foo = Vue.observable({ key: value });
> ```
> 
> _Update_: The above seems to work fine for me.

actually, you no need to use Vue.observable to create a observer object. just let foo = {key: value}.

**Comment by latel** [comment]
any progress?

**Comment by Cyclodex** [comment]
I think it is fixed now in latest: 9.1.2, isn't it?
At least in my project it looks better now...

**Comment by marked42** [comment]
This issue is fixed since v9.0.2. Refer to this https://github.com/kaorun343/vue-property-decorator/blob/v9.0.2/src/vue-property-decorator.ts#L69.
```ts
function produceProvide(original: any) {
  let provide: provideFunc = function (this: any) {
    let rv = typeof original === 'function' ? original.call(this) : original
    rv = Object.create(rv || null)
    // set reactive services (propagates previous services if necessary)
    rv[reactiveInjectKey] = Object.create(this[reactiveInjectKey] || {})
    for (let i in provide.managed) {
      rv[provide.managed[i]] = this[i]
    }
    for (let i in provide.managedReactive) {
      rv[provide.managedReactive[i]] = this[i] // Duplicates the behavior of `@Provide`
      Object.defineProperty(rv[reactiveInjectKey], provide.managedReactive[i], {
        enumerable: true,
        get: () => this[i],
      })
    }
    return rv
  }
  provide.managed = {}
  provide.managedReactive = {}
  return provide
}
```

Provided reactive object is changed from 

```ts
rv[reactiveInjectKey] = this[reactiveInjectKey] || {}
```
to

```ts
rv[reactiveInjectKey] = Object.create(this[reactiveInjectKey] || {})
```

`rv[reactiveInjectKey]` actually refers to `VueComponent.options.inject` so it's always the same object got called by `Object.defineProperty` which causes this issue. 

After this fix, every time a newly created object is provided to avoid Object.defineProperty called multiples on same object.

By the way, it seems to fix another issue, but coincidentally fixes this issue too, which explains why this issue is still open.

**Comment by qisi007** [comment]
我将`vue-property-decorator`更新到`9.1.2`解决了此问题
